scripts/data/convert_sft_data_for_olmocore.py

```diff
@@ -0,0 +1,140 @@
+        help="Dataset mixer list [dataset_name, weight]",
+    )
+    parser.add_argument(
+        "--dataset_split", type=str, default="train", help="Dataset split to use"
+    )
+    parser.add_argument(
+        "--max_seq_length", type=int, default=4096, help="Maximum sequence length"
+    )
+    parser.add_argument(
+        "--num_examples",
+        type=int,
+        default=0,
+        help="Number of examples to process for debugging. 0 means process all examples.",
+    )
+    parser.add_argument(
+        "--output_dir", type=str, required=True, help="Output directory"
+    )
+    parser.add_argument(
+        "--visualize", action="store_true", help="Visualize first token sequence"
+    )
+
+    args = parser.parse_args()
+
+    tc = TokenizerConfig(
+        tokenizer_name_or_path=args.tokenizer_name_or_path,
+        chat_template_name=args.chat_template_name,
+        add_bos=args.add_bos,
+    )
+
+    # TODO: improve configurability of transform factory
+    transform_functions_and_args = [
+        ("sft_tulu_tokenize_and_truncate_v1", {"max_seq_length": args.max_seq_length}),
+        ("sft_tulu_filter_v1", {}),
+    ]
+
+    transform_functions = [func for func, _ in transform_functions_and_args]
+    transform_function_args = [args for _, args in transform_functions_and_args]
+
+    train_dataset = get_cached_dataset_tulu(
+        dataset_mixer_list=args.dataset_mixer_list,
+        dataset_mixer_list_splits=[args.dataset_split],
+        tc=tc,
+        dataset_transform_fn=transform_functions,
+        transform_fn_args=transform_function_args,
+        target_columns=TOKENIZED_SFT_DATASET_KEYS,
+    )
+
+    if args.visualize:
+        visualize_token(train_dataset[0][INPUT_IDS_KEY], tc.tokenizer)
+        print("Labels:", train_dataset[0][LABELS_KEY])
+
+    if args.num_examples > 0:
+        print(f"Selecting {args.num_examples} examples for debugging")
+        train_dataset = train_dataset.select(range(args.num_examples))
+
+    print("Collecting tokens from dataset...")
+    token_ids = []
+    labels = []
+    attention_mask = []
+    sample: Mapping[str, Any]
+    for sample in train_dataset:
+        token_ids.extend(sample[INPUT_IDS_KEY])
+        labels.extend(sample[LABELS_KEY])
+        attention_mask.extend(sample[ATTENTION_MASK_KEY])
+
+    print(f"Total tokens: {len(token_ids)}")
+    print(f"Maximum token ID: {max(token_ids)}")
+
+    print("Writing data to numpy files...")
+
+    # Create output directory if it doesn't exist
+    os.makedirs(args.output_dir, exist_ok=True)
+
+    def write_memmap(filename, data, dtype):
+        mmap = np.memmap(filename, mode="w+", dtype=dtype, shape=(len(data),))
+        mmap[:] = data
+        mmap.flush()
+        return mmap
+
+    write_memmap(f"{args.output_dir}/token_ids.npy", token_ids, np.uint32)
+    write_memmap(f"{args.output_dir}/labels.npy", labels, np.int32)
+    write_memmap(f"{args.output_dir}/attention_mask.npy", attention_mask, np.int32)
```

Comment:
I don't know how big these datasets are, but if they are large enough then it would make sense to write out multiple numpy files like we do for pretraining datasets.